src/storage/mutate/UpdateEdgeProcessor.cpp

```diff
@@ -158,30 +159,56 @@ kvstore::ResultCode UpdateEdgeProcessor::collectEdgesProps(
             auto&& v = value(std::move(res));
             edgeFilters_.emplace(propName, v);
         }
-        auto schema = std::const_pointer_cast<meta::SchemaProviderIf>(constSchema);
         updater_ = std::unique_ptr<RowUpdater>(new RowUpdater(std::move(reader), schema));
     } else if (insertable_) {
         resp_.set_upsert(true);
         int64_t ms = time::WallClock::fastNowInMicroSec();
         auto now = std::numeric_limits<int64_t>::max() - ms;
         key_ = NebulaKeyUtils::edgeKey(partId, edgeKey.src, edgeKey.edge_type,
                                        edgeKey.ranking, edgeKey.dst, now);
-        const auto constSchema = this->schemaMan_->getEdgeSchema(this->spaceId_,
-                                                                 edgeKey.edge_type);
-        if (constSchema == nullptr) {
+        const auto schema = std::dynamic_pointer_cast<const meta::NebulaSchemaProvider>(
+                this->schemaMan_->getEdgeSchema(this->spaceId_, edgeKey.edge_type));
+        if (schema == nullptr) {
             return kvstore::ResultCode::ERR_UNKNOWN;
         }
-        for (auto index = 0UL; index < constSchema->getNumFields(); index++) {
-            auto propName = std::string(constSchema->getFieldName(index));
-            OptVariantType value = RowReader::getDefaultProp(constSchema.get(), propName);
+        auto updater = std::make_unique<RowUpdater>(schema);
+
+        for (auto index = 0UL; index < schema->getNumFields(); index++) {
+            auto propName = std::string(schema->getFieldName(index));
+            auto findIter = std::find_if(updateItems_.cbegin(), updateItems_.cend(),
+                    [&propName](auto &item) { return item.prop == propName; });
+            OptVariantType value;
+            if (findIter == updateItems_.end()) {
```

Comment:
Could you add some comments for the changes.

Reply:
yeah